src/test/TestUtils.h

```diff
@@ -10,14 +10,22 @@
 namespace stellar
 {
 
-Application::pointer
-createTestApplication(VirtualClock& clock, Config const& cfg);
-
 namespace testutil
 {
 void setCurrentLedgerVersion(LedgerManager& lm, uint32_t currentLedgerVersion);
 }
 
+template <typename T = ApplicationImpl>
+Application::pointer
+createTestApplication(VirtualClock& clock, Config const& cfg)
```

Comment:
having those implementations in the header files makes me a bit sad (and will cause us to recompile way too many things when updating the implementation of the factories). What do you think of doing something like https://isocpp.org/wiki/faq/templates#templates-defn-vs-decl instead?

Reply:
it's also that we depend on the compiler being smart to not generate bloated code. I guess we'll see if the stellar-core binaries become too big we can try newer versions of the compilers.